### PR TITLE
Add logging for duplicate record trees

### PR DESCRIPTION
**Pull request by zdg2102**

### CODE REVIEW

File: recidiviz/persistence/database/database.py [hunk 1]
```diff
@@ -149,11 +150,21 @@ def _convert_and_normalize_record_trees(
     any duplicates created by how SQLAlchemy handles joins
     """
     converted_people: List[entities.Person] = []
-    ids: Set[int] = set()
+    count_by_id: Dict[int, int] = defaultdict(lambda: 0)
     for person in people:
-        if person.person_id not in ids:
+        if count_by_id[person.person_id] == 0:
```
Comment (terinpw) [inline recidiviz/persistence/database/database.py]: nit just use `not count_by_id[person.person_id]`

File: recidiviz/persistence/database/database.py [hunk 2]
```diff
@@ -149,11 +150,21 @@ def _convert_and_normalize_record_trees(
     any duplicates created by how SQLAlchemy handles joins
```
Comment (terinpw) [inline recidiviz/persistence/database/database.py]: Not really related to this PR, but can we update this comment to say `remove any duplicate people created...`. From the method description it sounds like we're looking at duplicates for all entities, but in reality we're only checking people .

Comment (zdg2102) [reply recidiviz/persistence/database/database.py]: Done

File: recidiviz/persistence/database/database.py (hunk 1 above)
Comment (zdg2102) [reply recidiviz/persistence/database/database.py]: Discussed offline, will leave as is

**Pull request merged**